.github/workflows/clang-format-check.yml

```diff
@@ -0,0 +1,12 @@
+name: clang-format Check
+on: [push, pull_request]
+jobs:
+  formatting-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Run clang-format style check.
+      uses: DoozyX/clang-format-lint-action@v0.16.2
+      with:
+        exclude: './tpls'
```

Comment:
Did we need to exclude because of mdspan?  I expected we had disabled it for desul.
In any case that is fine we don't need to handle it now.

Reply:
That was my expectation as well but the first commit still needed it for some reason. I didn't check again for the second one but just kept it to be safe.

Reply:
Seems to work without it.